aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -32,6 +38,28 @@ Tensor& fill_(Tensor& self, const Tensor& value) {
   return self._fill_(value);
 }
 
+Tensor mvlgamma(const Tensor& self, int64_t p) {
+  AT_CHECK(at::isFloatingType(self.type().scalarType()),
+           "mvlgamma is not implemented for ", self.type());
+  AT_CHECK((self > 0.5 * (p - 1.)).all().toCByte(),
+           "Condition for computing multivariate log-gamma not met");
+  AT_CHECK(p >= 1, "p has to be greater than or equal to 1");
+  Tensor args = native::arange(-p + 1, 1, self.options()).div_(2.);
+  args = args.add(self.unsqueeze(-1));
```

Comment:
`add_`  seems better?

Reply:
I recall facing a problem with `add_` (a broadcasting issue), which is why I resorted to this.

Reply:
I see. How about `add_(self.unsqueeze(-1).expand_as(arg))s`?

Reply:
That solution is perhaps incorrect because it would effectively make `self` the same size as `args`.

Reply:
ah i see. it's the `args` that should be broadcasted, not `self.